test/unit_test/test_engine.py

```diff
@@ -44,11 +50,45 @@ def test_register(self):
         engine.register(p, host=system, evaluator=evaluator)
 
         # assert
-        assert (name, p) in engine._passes.items()
-        assert name in engine._pass_order
-        assert (name, system) in engine.hosts.items()
-        assert (name, evaluator) in engine._evaluators.items()
-        assert (name, False) in engine._clean_pass_run_cache.items()
+        assert name in engine.passes
+        assert name in engine.pass_order
+        assert engine.passes[name]["pass"] == p
+        assert engine.passes[name]["host"] == system
+        assert engine.passes[name]["evaluator"] == evaluator
+        assert engine.passes[name]["clean_run_cache"] is False
+
+    def test_register_no_search(self):
+        # setup
+        p = get_onnx_dynamic_quantization_pass(default_to_search=False)
+        name = p.__class__.__name__
+
+        options = {
+            "search_strategy": None,
+        }
```

Comment:
- If "search_strategy" is not provided or set to None then run-no-search path.
- If "search_strategy" is True then run the default search strategy
- If "search_strategy" is a dict then validate and run the strategy specified in the dict.

No need to create yet another option for the user to select default search strategy.

Reply:
This sounds like a good idea. 

what do you think the default search algorithm should be? `exhaustive`?

Reply:
yes

Reply:
Also, what should the default value for `search_strategy` be? `True` or `None`?